Pod/Classes/WPInputMediaPickerViewController.m

```diff
@@ -68,19 +67,69 @@ - (void)setupMediaPickerViewController {
 
 - (void)viewDidLayoutSubviews {
     [super viewDidLayoutSubviews];
+    [self configureCollectionView];
+}
 
-    CGFloat spacing = 1.0f;
-    CGFloat size = floorf((self.view.frame.size.height - spacing) / 2.0);
-    self.mediaPicker.options.cameraPreviewSize = CGSizeMake(1.5*size, 1.5*size);
+- (void)configureCollectionView {
+    CGFloat frameWidth = self.view.frame.size.width;
+    NSUInteger numberOfPhotosForLine = [self numberOfPhotosPerRow:frameWidth];
+    CGFloat photoSpacing = 1.0f;
+    CGFloat topInset = 5.0f;
+    CGFloat bottomInset = 10.0f;
+
+    CGFloat cellSize = [self.mediaPicker cellSizeForPhotosPerLineCount:numberOfPhotosForLine
+                                                          photoSpacing:photoSpacing
+                                                            frameWidth:frameWidth];
+
+    // Check the actual width of the content based on the computed cell size
+    // How many photos are we actually fitting per line?
+    CGFloat totalSpacing = (numberOfPhotosForLine - 1) * photoSpacing;
+    numberOfPhotosForLine = floorf((frameWidth - totalSpacing) / cellSize);
+
+    CGFloat contentWidth = (numberOfPhotosForLine * cellSize) + totalSpacing;
+
+    // If we have gaps in our layout, adjust to fit
+    if (contentWidth < frameWidth) {
+        cellSize = [self.mediaPicker cellSizeForPhotosPerLineCount:numberOfPhotosForLine
+                                                      photoSpacing:photoSpacing
+                                                        frameWidth:frameWidth];
+    }
+    
+    // Init and configure collection view layout
     UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc] init];
-    layout.scrollDirection = UICollectionViewScrollDirectionHorizontal;
-    layout.itemSize = CGSizeMake(size, size);
-    layout.minimumLineSpacing = spacing;
-    layout.minimumInteritemSpacing = spacing;
-    layout.sectionInset = UIEdgeInsetsMake(0, 5, 0, 10);
+    layout.scrollDirection = UICollectionViewScrollDirectionVertical;
+    layout.itemSize = CGSizeMake(cellSize, cellSize);
+    layout.minimumInteritemSpacing = photoSpacing;
+    layout.minimumLineSpacing = photoSpacing;
+    layout.sectionInset = UIEdgeInsetsMake(topInset, 0, bottomInset, 0);
 
+    self.mediaPicker.options.cameraPreviewSize = CGSizeMake(1.5*cellSize, 1.5*cellSize);
     self.mediaPicker.collectionView.collectionViewLayout = layout;
+}
 
+/**
+ Returns the maximum number of photos to be used in a picker row given the provided frame width.
+ 
+ @param frameWidth Width size of the frame containing the picker
+
+ @return The number of photo cells to be used in a row. Defaults to 3.
+ */
+- (NSUInteger)numberOfPhotosPerRow:(CGFloat)frameWidth {
+    NSUInteger numberOfPhotos = 3;
+
+    if (frameWidth >= 375 && frameWidth < 500) {
```

Comment:
@SergioEstevao and @kwonye I felt the same way as I wrote this. Size classes would work for some of this, however, there are circumstances where they would not work. An example would be the iPhone SE and iPhone 7+ in portrait — both are considered "compact width". Per the designs, we needed a little more granular control over the precise width of the frame (e.g. 3-wide for the SE and 4-wide for the 7/7+), hence the frame sizes here. 

I can look into this a little more today.